core/src/main/java/hudson/cli/declarative/OptionHandlerExtension.java

```diff
@@ -35,7 +35,6 @@
 
 /**
  * {@link OptionHandler}s that should be auto-discovered.
- * TODO is this actually necessary? {@code @MetaInfServices(OptionHandler.class)} seems to work as well.
```

Comment:
Answer: it is actually necessary if we want to remove Commons Discovery someday, as `ServiceLoader` cannot work with the `OptionHandler` classes that have no public zero-argument constructor.